verification/store.go

```diff
@@ -75,3 +70,33 @@ func LoadX509TrustStore(path string) (*X509TrustStore, error) {
 
 	return &trustStore, nil
 }
+
+func validateCerts(certs []*x509.Certificate, path string) error {
+	// to prevent any trust store misconfigurations, ensure there is at least
+	// one certificate from each file.
+	if len(certs) < 1 {
+		return fmt.Errorf("could not parse a certificate from %q, every file in a trust store must have a PEM or DER certificate in it", path)
+	}
+
+	if len(certs) == 1 {
+		// if there is only one certificate, it must be a self-signed cert or
```

Comment:
Single certificate in trust store doesn't means that its a self signed signing certificate. Trust store can have multiple certificate and one out of them can be a self signed signing certificate.

Logic:
```go
func validateCerts(certs[] * x509.Certificate, path string) error {
    // to prevent any trust store misconfigurations, ensure there is at least
    // one certificate from each file.
    if len(certs) < 1 {
        return fmt.Errorf("could not parse a certificate from %q, every file in a trust store must have a PEM or DER certificate in it", path)
    }
    for _, cert: = range certs {
        if !cert.IsCA {
                if err: = cert.CheckSignature(cert.SignatureAlgorithm, cert.RawTBSCertificate, cert.Signature);
                err != nil {
                    fmt.Errorf("certificate with subject %q from file %q is not a CA certificate or self-signed signing certificate", cert.Subject, joinedPath)
                }
            }
        }
    }
}

```

Reply:
thanks for the comment! Updated.

Reply:
Actually, I found https://pkg.go.dev/crypto/x509#Certificate.CheckSignatureFrom might better fit this case, and it uses `CheckSignature` under the hood. Let me know if you have any concerns on it. thanks! @priteshbandi

Reply:
We cannot use [CheckSignatureFrom](https://pkg.go.dev/crypto/x509#Certificate.CheckSignatureFrom) because it imposes some extra constraints that are not valid. Instead we should use CheckSignature